modules/integration/tests/src/test/java/org/wso2/appmanager/integration/test/cases/publisher/webapp/lifecycle/management/CreateWebAppTestCase.java

```diff
@@ -0,0 +1,171 @@
+        HttpResponse webAppAddedResponse = publisherRestClient.addWebApp(webApp);
+        JSONObject webAppAddedResponseData = new JSONObject(webAppAddedResponse.getData());
+        //Test whether app is created successfully.
+        assertEquals(webAppAddedResponseData.getString(AppmTestConstants.OK), "true", "Creating web application is " +
+                " not allowed for user : " + userName + " who has sufficient privileges to create WebApp.");
+        assertEquals(webAppAddedResponseData.getString(AppmTestConstants.MESSAGE), "asset added", "Creating WebApp " +
+                "is not allowed for user : " + userName + " who has sufficient privileges to create WebApp.");
+        String appId = webAppAddedResponseData.getString(AppmTestConstants.ID);
+        assertNotNull(appId, "user " + userName + " who has sufficient privileges to create WebApp, failed to create" +
+                " WebApp");
+
+        addTagsAndRoles(publisherRestClient, webApp);
+        //Add a service provider to the created WebApp.
+        HttpResponse ssoProviderAddedResponse = publisherRestClient.addSsoProvider(webApp);
+        int responseCode = ssoProviderAddedResponse.getResponseCode();
+        assertTrue(responseCode == 200, "Excepted status code is 200 for user :" + userName + ". But received status " +
+                "code is responseCode");
+        //Logout from publisher by valid user.
+        publisherRestClient.logout();
+        appIdList.add(appId);
+    }
+
+    @Test(dataProvider = "inValidUserModeDataProvider", description = TEST_DESCRIPTION)
+    public void testCreateWebAppWithInValidUsers(User user, String prefix) throws Exception {
+        APPMPublisherRestClient publisherRestClient = new APPMPublisherRestClient(backEndUrl);
+        // Login to publisher by invalid user.
+        String userName = user.getUserName();
+        publisherRestClient.login(userName, user.getPassword());
+        WebApp webApp = createWebAppObject(publisherRestClient, user, prefix);
+        HttpResponse webAppAddedResponse = publisherRestClient.addWebApp(webApp);
+        JSONObject webAppAddedResponseData = new JSONObject(webAppAddedResponse.getData());
+        //Logout from publisher by invalid user.
+        publisherRestClient.logout();
+        assertEquals(webAppAddedResponseData.getString(AppmTestConstants.OK), "false", "Creating web application is " +
+                "allowed for user : " + userName + " who has insufficient privileges to create WebApp.");
+        assertNotEquals(webAppAddedResponseData.getString(AppmTestConstants.MESSAGE), "asset added", "Creating WebApp " +
+                "is allowed for user : " + userName + " who has insufficient privileges to create WebApp.");
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void closeDown() throws Exception {
+        //delete all created apps by admin.
+        for (int i = 0; i < appIdList.size(); i++) {
+            appmPublisherRestClient.deleteApp(appIdList.get(i));
```

Comment:
@lahiru823 , As I can see, the code just iterates a "List of AppID"s. Hence I think it is correct to start form 0. Otherwise very first app added in the test might not get removed.